corehq/apps/userreports/reports/builder/forms.py

```diff
@@ -102,19 +102,25 @@ def render(self, name, value, attrs=None, choices=()):
         final_attrs = self.build_attrs(attrs, name=name)
 
         return format_html(
-            '<input{0} data-bind="'
-            '   questionsSelect: {1},'
-            '   value: \'{2}\','
-            '   optionsCaption: \' \''
-            '"/>',
+            """
+            <input{0} data-bind='
+               questionsSelect: {1},
+               value: "{2}",
+               optionsCaption: " "
+            '/>
+            """,
             flatatt(final_attrs),
             mark_safe(self.render_options(choices)),
             value
         )
 
     def render_options(self, choices):
+
+        def escape(literal):
+            return literal.replace('&', '&amp;').replace("'", "&#39;")
```

Comment:
I definitely would prefer a library function, but I couldn't find one that does quite what I want (although I suspect one exists). This string is going to be used in the above single-quoted data-bind attribute. Based on the [HTML spec](http://www.w3.org/TR/html5/syntax.html#attribute-value-%28single-quoted%29-state), it looks like ampersand and single quote are the only characters that need escaping in the "attribute value (single-quoted) state".
(`cgi.escape` escapes `&`, `<`, `>`, and optionally `"`, but not `'`)